Sources/Commands/SwiftTool.swift

```diff
@@ -203,6 +204,17 @@ public class SwiftTool<Options: ToolOptions> {
         if let workspace = _workspace {
             return workspace
         }
+        let envSearchPaths = Utility.getEnvSearchPaths(
+            pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
+
+        guard let gitLocation = Utility.lookupExecutablePath(filename: Git.tool, searchPaths: envSearchPaths) else {
+            throw SwiftToolError.missingGitDependency
+            // NOTE(heckj): differentiate error reporting between not finding it and it not being executable?
+        }
+        guard localFileSystem.isExecutableFile(gitLocation) else {
+            throw SwiftToolError.missingGitDependency
+        }
```

Comment:
I think this may be a preferrable solution, not because it's specific to git, but because it only happens once and the beginning of any invocation, regardless of how many times "git" is invoked - which in a larger project could be quite a number of times. Vapor lists 6 dependencies for example - 5 times more checks than is strictly needed to validate Git is around and in the path as an executable. 

If you insist that we should do this check as a part of Process, I'll make the changes and updates, but I'd like to ask that you reconsider.